sgt-back-end/index.js

```diff
@@ -0,0 +1,167 @@
+const express = require('express');
+const pg = require('pg');
+
+const app = express();
+app.use(express.json());
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.get('/api/grades', (req, res) => {
+  const sql = `
+    select "gradeId",
+           "name",
+           "course",
+           "score",
+           "createdAt"
+    from "grades"
+  `;
+  db.query(sql)
+    .then(result => {
+      const grades = result.rows;
+      res.status(200).json(grades);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.post('/api/grades', (req, res) => {
+  const newStudent = req.body;
+  if (!(Number.isInteger(newStudent.score)) || newStudent.score < 0 || newStudent.score > 100) {
+    res.status(400).json({
+      error: '"score" must be a positive integer between 0 and 100'
+    });
+  } else if (!newStudent.name) {
+    res.status(400).json({
+      error: '"name" is a required field'
+    });
+  } else if (!newStudent.course) {
+    res.status(400).json({
+      error: '"course" is a required field'
+    });
+  } else if (!newStudent.score) {
+    res.status(400).json({
+      error: '"score" is a required field'
+    });
+  } else {
+    const params = [];
+    for (const key in newStudent) {
+      params.push(newStudent[key]);
+    }
```

Comment:
It's not a good idea for your code to make assumptions about property order. You can populate your params manually.

```js
const params = [newStudent.name, newStudent.course, newStudent.score];
```